src/subscript/welltest_extract/welltest_extract.py

```diff
@@ -0,0 +1,396 @@
+#!/usr/bin/env python
+
+import sys
+import argparse
+import numpy as np
+from ecl.summary import EclSum
+
+
+DESCRIPTION = """
+Script to extract simulated welltest results from simulator output.
+Outputs pressure vs superpositioned time derivative
+
+Required summary vectors in sim deck:
+  -days,
+  -wbhp:well_name,
+  -wwpr:well_name,
+  -wopr:well_name if main_phase == OIL
+  -wgpr:well_name if main_phase == GAS
+"""
+
+"""
+
+Script is a rewrite of a legacy script originally developed and improved by:
+ * Jon Saetrom
+ * Bjoern Kaare Hegstad
+ * Cecile Otterlei
+ * Hodjat Moradi
+
+AUTHOR: Eivind Smoergrav
+
+TODO
+ - Support pseudo pressure vs time relevant for gas and gas condensate fields.
+ - Make error handling more robust, eg:
+     - check if all necessary vectors are present wbhp, wopr or wgpr, wwpr is optional
+- decide how the output files should be handled, there are many and messy as of now
+- make an ERT JOB
+- submit to subscript and ert handle in komodo
+- old script did not exist(1) with invalid input, but reported nans in the outfiles,
+  which is very bad practice. Discuss with users
+"""
+
+
+def get_parser():
+    """
+    Define the argparse parser
+    """
+    parser = argparse.ArgumentParser(
+        description=DESCRIPTION,
+        formatter_class=argparse.RawTextHelpFormatter,
+    )
+
+    parser.add_argument(
+        "eclcase",
+        type=str,
+        help="Eclipse case to extract from",
+    )
+    parser.add_argument(
+        "well_name",
+        type=str,
+        help="Name of well to extract results from",
+    )
+    parser.add_argument(
+        "out_files_prefix",
+        type=str,
+        help="Prefix to be added to result files, can contain a path",
+    )
+    parser.add_argument(
+        "-n",
+        "--buildup_nr",
+        type=int,
+        help="Buildup number, indicating which build up to extract. Counting from 1 ",
+        default=1,
+        required=False,
```

Comment:
An option is by default not required, so `required=False` can be dropped.

Reply:
Fixed